test/integration/targets/elb_application_lb/tasks/main.yml

```diff
@@ -0,0 +1,524 @@
+- block:
+
+  - name: set connection information for all tasks
+    set_fact:
+      aws_connection_info: &aws_connection_info
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        region: "{{ aws_region }}"
+    no_log: yes
+
+  - name: load balancer name has to be less than 32 characters
+    # the 8 digit identifier at the end of resource_prefix helps determine during which test something
+    # was created and allows tests to be run in parallel
+    set_fact:
+      alb_name: "my-alb-{{ item }}"
+    with_items: "{{ resource_prefix | regex_findall('.{8}$') }}"
+
+  - name: tg name has to be less than 32 characters
+    set_fact:
+      tg_name: "my-tg-{{ item }}"
+    with_items: "{{ resource_prefix | regex_findall('.{8}$') }}"
+
+  - name: create VPC
+    ec2_vpc_net:
+      cidr_block: 10.228.228.0/22
+      name: "{{ resource_prefix }}_vpc"
+      state: present
+      <<: *aws_connection_info
+    register: vpc
+
+  - name: create internet gateway
+    ec2_vpc_igw:
+      vpc_id: "{{ vpc.vpc.id }}"
+      state: present
+      <<: *aws_connection_info
+    register: igw
+
+  - name: create public subnet
+    ec2_vpc_subnet:
+      cidr: "{{ item.cidr }}"
+      az: "{{ aws_region}}{{ item.az }}"
+      vpc_id: "{{ vpc.vpc.id }}"
+      state: present
+      tags:
+        Public: "{{ item.public|string }}"
+        Name: "{{ item.public|ternary('public', 'private') }}-{{ item.az }}"
+      <<: *aws_connection_info
+    with_items:
+      - cidr: 10.228.228.0/24
+        az: "a"
+        public: "True"
+      - cidr: 10.228.229.0/24
+        az: "b"
+        public: "True"
+      - cidr: 10.228.230.0/24
+        az: "a"
+        public: "False"
+      - cidr: 10.228.231.0/25
+        az: "b"
+        public: "False"
+      - cidr: 10.228.231.128/25
+        az: "c"
```

Comment:
Do we need to use this third zone (I'm trying to keep the tests region independent and I know that not all regions have more than two zones)

Reply:
Removed :)